src/typescript-reporter/reporter/ControlledTypeScriptSystem.ts

```diff
@@ -97,24 +102,42 @@ function createControlledTypeScriptSystem(
       return;
     }
 
-    const directoryWatchers = directoryWatchersMap.get(directory);
-    if (directoryWatchers) {
-      directoryWatchers.forEach((directoryWatcher) =>
-        directoryWatcher(forwardSlash(normalizedPath))
+    const directoryWatcherCallbacks = directoryWatcherCallbacksMap.get(directory);
+    if (directoryWatcherCallbacks) {
+      directoryWatcherCallbacks.forEach((directoryWatcherCallback) =>
+        directoryWatcherCallback(forwardSlash(normalizedPath))
       );
     }
 
-    recursiveDirectoryWatchersMap.forEach((recursiveDirectoryWatchers, watchedDirectory) => {
-      if (
-        watchedDirectory === directory ||
-        (directory.startsWith(watchedDirectory) &&
-          forwardSlash(directory)[watchedDirectory.length] === '/')
-      ) {
-        recursiveDirectoryWatchers.forEach((recursiveDirectoryWatcher) =>
-          recursiveDirectoryWatcher(forwardSlash(normalizedPath))
-        );
+    recursiveDirectoryWatcherCallbacksMap.forEach(
+      (recursiveDirectoryWatcherCallbacks, watchedDirectory) => {
+        if (
+          watchedDirectory === directory ||
+          (directory.startsWith(watchedDirectory) &&
+            forwardSlash(directory)[watchedDirectory.length] === '/')
+        ) {
+          recursiveDirectoryWatcherCallbacks.forEach((recursiveDirectoryWatcherCallback) =>
+            recursiveDirectoryWatcherCallback(forwardSlash(normalizedPath))
+          );
+        }
       }
-    });
+    );
+  }
+
+  function updateDirectorySnapshot(path: string, recursive = false) {
+    const dirents = passiveFileSystem.readDir(path);
```

Comment:
What is a `dirent`? Is that directory?

Reply:
It's a name from the Node.js API (there is a `Dirent` type) which seems to be borrowed from the POSIX API. It's a structure that represents directory children (can be another directory, file, device, etc.)